server/jetstream_cluster.go

```diff
@@ -2439,16 +2438,15 @@ func (js *jetStream) monitorStream(mset *stream, sa *streamAssignment, sendSnaps
 	// a complete and detailed state which could be costly in terms of memory, cpu and GC.
 	// This only entails how many messages, and the first and last sequence of the stream.
 	// This is all that is needed to detect a change, and we can get this from FilteredState()
-	// with and empty filter.
+	// with an empty filter.
 	var lastState SimpleState
-	var lastSnapTime time.Time
 
 	// Don't allow the upper layer to install snapshots until we have
 	// fully recovered from disk.
 	isRecovering := true
 
 	doSnapshot := func() {
-		if mset == nil || isRecovering || isRestore || time.Since(lastSnapTime) < minSnapDelta {
+		if mset == nil || isRecovering || isRestore {
```

Comment:
This was here to not snaphot faster then every 10s, not to snapshot every 10s..

Reply:
Indeed, a snapshot will happen either every ~2-3 minutes or (earlier) when compaction limits are reached. Previously that was with a maximum of 10s between subsequent snapshots being taken, but it turns out that has worse performance than doing the snapshotting when needed without this limit.

This 10s was essentially ignored before due to the bug that was fixed in https://github.com/nats-io/nats-server/pull/6277, and it turns out not using those 10s actually has better performance.